packages/live-preview-sdk/src/inspectorMode/utils.ts

```diff
@@ -53,21 +58,87 @@ export function getInspectorModeAttributes(
 /**
  * Query the document for all tagged elements
  */
-export function getAllTaggedElements(root = window.document): Element[] {
+export function getAllTaggedElements(root = window.document, ignoreManual?: boolean): Element[] {
   // The fastest way to look up & iterate over DOM. Ref:
   // https://stackoverflow.com/a/2579869
   //
   // Terminology:
   // FILTER_SKIP: Skip the current node
   // FILTER_REJECT: Skip the current node and all its children
   const walker = document.createTreeWalker(root, NodeFilter.SHOW_ALL, (node) => {
+    if (node.nodeType === Node.TEXT_NODE) {
+      const text = node.textContent ?? '';
+
+      const { origin, href } = (vercelStegaDecode(text) ?? {}) as Record<string, string>;
+
+      if (origin !== 'contentful.com' || !href) {
+        return NodeFilter.FILTER_SKIP;
+      }
+
+      return isTaggedElement(node.parentElement)
+        ? ignoreManual
+          ? NodeFilter.FILTER_ACCEPT
+          : NodeFilter.FILTER_SKIP
+        : NodeFilter.FILTER_ACCEPT;
+    }
+
+    if (ignoreManual) {
+      return NodeFilter.FILTER_SKIP;
+    }
+
     return isTaggedElement(node) ? NodeFilter.FILTER_ACCEPT : NodeFilter.FILTER_SKIP;
   });
 
   const elements: Element[] = [];
 
   while (walker.nextNode()) {
-    elements.push(walker.currentNode as Element);
+    const node = walker.currentNode;
+
+    if (node.nodeType === Node.ELEMENT_NODE) {
+      elements.push(walker.currentNode as Element);
+      continue;
+    }
+
+    // Handle Encoded strings
+    const { href } = (vercelStegaDecode(node.textContent ?? '') ?? {}) as Record<string, string>;
+    const el = node.parentElement;
+
+    if (!el) {
+      console.log('skipping no parent');
+      continue;
+    }
+
+    // FIXME: maybe we should encode these attributes as separate attributes
```

Comment:
any objection for adding these info as separate attributes?

Reply:
You mean not parsing them from the URL and having them directly as properties?

```
const { spaceId, environmentId, entryId, assetId, fieldId, localeCode } = vercelStegaDecode(node.textContent).data
```

Only con I would see is that the hidden data would be even more payload and it could affect the loading time (more) than having the information only once

Reply:
> You mean not parsing them from the URL and having them directly as properties?

Exactly.

Reply:
As mentioned the only con I would see is regarding performance
But I would be also in favor of having the data directly without the additional need for parsing, makes it more clearer and easier to understand/build on

Reply:
I agree. But also, with the payload size it may not be big of an issue since it's only enabled for preview deployments.